### PR TITLE
Fix selection of multiword

### PR DESCRIPTION
**Pull request by guillep**
A string of N characters have N+1 caret positions. `#nextWord:` should take that into account.

Fix forwardDelete: also for mac machines

### CODE REVIEW

**Comment by jecisc** [comment]
We have failing tests:

osx-64 / Tests-osx-64 / testPreviousWord – MacOSX64.Rubric.Tests.Editing.Core.RubTextEditorTest<1s
osx-64 / Tests-osx-64 / testSelectWordMarkPoint – MacOSX64.Rubric.Tests.Editing.Core.RubTextEditorTest<1s

They are present in the 3 OS

**Comment by jecisc** [comment]
Oh, a fix was pushed right when I commented ahah

File: src/NECompletion-Tests/CompletionEngineTest.class.st [hunk 1]
```diff
@@ -229,18 +229,18 @@ CompletionEngineTest >> testReplaceKeywordTokenWithCaretInTheEndOfWordWithFollow
...
 
 	editor textArea openInWorld.
 	controller openMenu.
 
-	controller context replaceTokenInEditorWith: 'mEthOdThatDoesNotExist:'.
+	controller context replaceTokenInEditorWith: 'toto'.
 
-	self assert: editor text asString equals: 'self mEthOdThatDoesNotExist: something that follows'
+	self assert: editor text asString equals: 'toto'
 ]
```
Comment (PalumboN) [inline src/NECompletion-Tests/CompletionEngineTest.class.st]: Take care with this test, we changed the semantics.
I think that we should leave the test as it was and add a new one with the new example.

**Review by PalumboN — COMMENTED** [review]
Good tests! I left one comment about one of them 😇

**Pull request merged**